File: backend/src/schemas/transaction.py
"""
Transaction domain model.
"""
from datetime import date, datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field


class Transaction(BaseModel):
    """A financial transaction (income or expense)."""

    id: UUID
    user_id: UUID
    amount: int = Field(..., description="Amount in cents (fen)")
    category_id: UUID
    note: str = ""
    date: date
    type: Literal["income", "expense"]
    trade_no: str = ""
    created_at: datetime

    model_config = {"str_strip_whitespace": True}

File: frontend/src/schemas/transaction.ts
export type TransactionType = "income" | "expense";

export interface Transaction {
  id: string;
  amount: number;
  category_id: string;
  note: string;
  date: string;
  type: TransactionType;
  created_at: string;
}

export interface TransactionCreate {
  amount: number;
  category_id: string;
  note?: string;
  date: string;
  type: TransactionType;
}

export interface TransactionUpdate {
  category_id?: string;
  note?: string;
}

export interface TransactionFilter {
  start_date?: string;
  end_date?: string;
  category_id?: string;
  type?: TransactionType;
}
